promql/functions.go

```diff
@@ -184,6 +184,107 @@ func funcIrate(ev *evaluator, args Expressions) model.Value {
 	return resultVector
 }
 
+// doubleSVal calculates the smoothed value at index i of raw data d
```

Comment:
This comment makes no sense unless you already understand what this function does. Can you add some context?